tests/unit/components/panel/AttributeSection.spec.js

```diff
@@ -0,0 +1,325 @@
+          name: 'attribute',
+          type: 'String',
+          value: [{
+            name: 'attribute_1',
+            type: 'String',
+            value: '',
+            definition: null,
+          }],
+        },
+      }]]);
+    });
+  });
+
+  describe('Test function: deleteAttribute', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.deleteAttribute();
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'attribute',
+      }]]);
+    });
+  });
+
+  describe('Test function: updateAttributeName', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.updateAttributeName('test');
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'test',
+        attribute: {
+          name: 'test',
+          type: 'String',
+          value: [],
+        },
+      }]]);
+    });
+  });
+
+  describe('Test function: updateAttributeValue', () => {
+    it('should emit event', async () => {
+      await wrapper.vm.updateAttributeValue('test');
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'attribute',
+        attribute: {
+          name: 'attribute',
+          type: 'String',
+          value: 'test',
+        },
+      }]]);
+    });
+  });
+
+  describe('Test function: updateAttribute', () => {
+    it('should remove attribute when call function with event without attribute', async () => {
+      await wrapper.setProps({
+        attribute: {
+          name: 'test',
+          value: [{
+            name: 'attribut_1',
+            value: '',
+            definition: null,
+            type: 'String',
+          }],
+        },
+      });
+
+      wrapper.vm.updateAttribute({ name: 'attribut_1' });
+
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'test',
+        attribute: {
+          name: 'test',
+          value: [],
+        },
+      }]]);
+    });
+
+    it('should update attribute value when call function with event with existing attribute name', async () => {
+      await wrapper.setProps({
+        attribute: {
+          name: 'test',
+          value: [{
+            name: 'attribut_1',
+            value: '',
+            definition: null,
+            type: 'String',
+          }],
+        },
+      });
+
+      wrapper.vm.updateAttribute({
+        name: 'attribut_1',
+        attribute: {
+          name: 'attribut_2',
+          value: 'test',
+          definition: null,
+          type: 'String',
+        },
+      });
+
+      expect(wrapper.emitted()['update:attribute']).toEqual([[{
+        name: 'test',
+        attribute: {
+          name: 'test',
+          value: [{
+            name: 'attribut_2',
+            value: 'test',
+            definition: null,
+            type: 'String',
+          }],
+        },
+      }]]);
+    });
+
+    it('should add attribute value when call function with event without existing attribute name', async () => {
```

Comment:
with event with attribute name not existing

Reply:
done